src/app/forum/containers/thread/thread.component.html

```diff
@@ -1,119 +1,128 @@
-            <form class="send-form" [formGroup]="form" (ngSubmit)="sendMessage(threadId, isThreadOpened)">
-              <textarea
-                class="textarea"
-                placeholder="Send a message..."
-                formControlName="messageToSend"
-                [rows]="1"
-                algAutoResize
-                (keydown.enter)="$event.preventDefault(); sendMessage(threadId, isThreadOpened)"
-                #messageToSendEl
-              ></textarea>
-              <button
-                class="size-l with-bg send-button"
-                type="submit"
-                icon="pi pi-send"
-                alg-button-icon
-                [disabled]="!form.get('messageToSend')?.value?.trim() || !!(disableControls$ | async)"
-              ></button>
-            </form>
-            <div class="thread-status">
-              <div class="thread-status-caption" >
-                <ng-container *ngIf="thread.threadStatus?.data?.open">
-                  <span i18n>This thread is open</span>
-                </ng-container>
-                <ng-container *ngIf="thread.threadStatus?.data?.open === false">
-                  <span i18n>This thread is closed</span>
-                </ng-container>
-                <ng-container *ngIf="thread.threadStatus?.isFetching">
-                  <span i18n>Loading thread status...</span>
-                </ng-container>
-                <ng-container *ngIf="thread.threadStatus?.isError">
-                  <span i18n>Error while loading thread info</span>
-                </ng-container>
-                <ng-container *ngIf="thread.threadStatus === undefined">
-                  <span i18n>Could not load thread</span>
-                </ng-container>
-              </div>
-              <div
-                *ngrxLet="!!thread.threadStatus && !!thread.threadStatus.data && ((thread.threadStatus.data.open && thread.threadStatus.data.canClose) || (!thread.threadStatus.data.open && thread.threadStatus.data.canOpen)) as canSwitch"
-                [pTooltip]="threadStatusTooltip"
-                [tooltipDisabled]="canSwitch"
-                tooltipPosition="top"
-                tooltipStyleClass="alg-tooltip"
-              >
-                <button
-                  type="button"
-                  [disabled]="!canSwitch"
-                  alg-button
-                  (click)="changeThreadStatus({ open: !thread.threadStatus.data.open, threadId: threadId })"
-                  *ngIf="thread.threadStatus && thread.threadStatus.data"
-                >
-                  <span *ngIf="thread.threadStatus.data.open; else openThreadCaption" i18n>Close this thread</span>
-                  <ng-template #openThreadCaption>
-                    <span i18n>Open this thread</span>
+            } @else {
+              @if (threadMetadata.id) {
+                @for (event of state().data; track event) {
```

Comment:
As the `mergeEvents` returns new array, I assume that the track by event (so link) won't work here

Reply:
The content of the array would be the same objects (as long as there is not a new refresh)... so I assume it should work.
But I can use the `time` as well maybe.